src/NUnitCore/core/CoreExtensions.cs

```diff
@@ -221,12 +221,22 @@ public void InstallAddins()
 			}
 		}
 
+        public ArrayList AdhocExtensions = new ArrayList();
+
 		public void InstallAdhocExtensions( Assembly assembly )
 		{
+            AdhocExtensions.Clear();
+
 			foreach ( Type type in assembly.GetExportedTypes() )
 			{
-				if ( type.GetCustomAttributes(typeof(NUnitAddinAttribute), false).Length == 1 )
-					InstallAddin( type );
+                if (type.GetCustomAttributes(typeof(NUnitAddinAttribute), false).Length == 1)
```

Comment:
`if (type.IsDefined(typeof(NUnitAddinAttribute), false))`?

Reply:
Good. This is one of the few places where we can use actual reflection rather than a `Reflect` call, since `NUnitAddinAttribute` is defined in core.